### PR TITLE
Task 2 implemented

### PR DESCRIPTION
**Pull request by varivoda123**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
```
Comment (a-bart) [inline webpack.config.js:1]: It's not necessary, but I would suggest to split dev and prod configuration into separate files. As in future config will grow and it'll become hardly to read.

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          test: /\.jsx?$/,
+          exclude: /node_modules/,
+          loader: 'babel-loader',
+          options: {
+            presets: ['@babel/preset-env', '@babel/preset-react'],
+          },
+        },
+        {
+          test: /\.css$/,
+          use: [MiniCssExtractPlugin.loader, 'css-loader'],
```
Comment (a-bart) [inline webpack.config.js:31]: it's better to use `MiniCssExtractPlugin.loader` for prod, and `style-loader` for dev

File: webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+const OptimizeCssAssetsPlugin = require('optimize-css-assets-webpack-plugin');
+const TerserPlugin = require('terser-webpack-plugin');
+const path = require('path');
+
+module.exports = (env) => {
+  const isProduction = env && env.production;
+
+  return {
+    mode: isProduction ? 'production' : 'development',
+    devtool: isProduction ? 'source-map' : 'inline-source-map',
```
Comment (a-bart) [inline webpack.config.js:13]: Can you add custom devServer configuration?
For example, to set custom port and automatically open application in browser on start.

File: webpack.config.js (hunk 1 above)
Comment (varivoda123) [reply webpack.config.js]: done

File: webpack.config.js (hunk 3 above)
Comment (varivoda123) [reply webpack.config.js]: done, added to dev configuration

File: webpack.config.js (hunk 2 above)
Comment (varivoda123) [reply webpack.config.js]: done

File: webpack.config.common.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    filename: '[name].[contenthash].js',
+  },
+  module: {
+    rules: [
+      {
+        test: /\.jsx?$/,
+        exclude: /node_modules/,
+        loader: 'babel-loader',
+        options: {
+          presets: ['@babel/preset-env', '@babel/preset-react'],
```
Comment (a-bart) [inline webpack.config.common.js:18]: `presets` could be moved into .babelrc file. Up to you

**Pull request merged**